api/project.py

```
@@ -0,0 +1,110 @@
+    
+    return path
+
+class Project():
+    def __init__(self, name: str, path: Path, url: Optional[str]):
+        self.url   = url
+        self.name  = name
+        self.path  = path
+        self.graph = Graph(name)
+
+        if url is not None:
+            save_repo_info(name, url)
```

Ruff (0.8.2)

`save_repo_info` may be undefined, or defined from star imports

(F405)